### PR TITLE
refactor: Deprecate transcript & triggers endpoints and create tests

### PR DESCRIPTION
**Pull request by KevLehman**
OC-990
OC-1005
OC-1012
OC-1008

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts [hunk 1]
```diff
@@ -277,6 +277,140 @@ describe('LIVECHAT - Utils', function () {
...
 		});
 	});
+	describe('livechat/transcript/:rid', () => {
+		it('should fail if user is not authenticated', async () => {
+			await request.delete(api('livechat/transcript/rid')).send({}).expect(401);
+		});
+		it('should fail if user doesnt have "send-omnichannel-chat-transcript" permission', async () => {
+			const user = await createVisitor();
+			const room = await createLivechatRoom(user.token);
+			await updatePermission('send-omnichannel-chat-transcript', []);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts:287]:
```suggestion
			await removePermissionFromAllRoles('send-omnichannel-chat-transcript');
```

File: apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts [hunk 2]
```diff
@@ -277,6 +277,140 @@ describe('LIVECHAT - Utils', function () {
...
+
+			await request
+				.delete(api(`livechat/transcript/${room._id}`))
+				.set(credentials)
+				.send({})
+				.expect(403);
+		});
+		it('should fail if rid is not a valid room id', async () => {
+			await updatePermission('send-omnichannel-chat-transcript', ['livechat-manager', 'admin']);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts:296]:
```suggestion
			await restorePermissionToRoles('send-omnichannel-chat-transcript');
```

File: apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts [hunk 3]
```diff
@@ -277,6 +277,140 @@ describe('LIVECHAT - Utils', function () {
...
+			await request.delete(api('livechat/transcript/rid')).set(credentials).send({}).expect(400);
+		});
+		it('should fail if room is not open', async () => {
+			const user = await createVisitor();
+			const room = await createLivechatRoom(user.token);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts:301]: Perhaps you can use `startANewLivechatRoomAndTakeIt` here which internally creates visitor and room

File: apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts [hunk 4]
```diff
@@ -277,6 +277,140 @@ describe('LIVECHAT - Utils', function () {
...
+			await closeOmnichannelRoom(room._id);
+			await request
+				.delete(api(`livechat/transcript/${room._id}`))
+				.set(credentials)
+				.send({})
+				.expect(400);
+		});
+		it('should fail if room doesnt have transcript requested', async () => {
+			const user = await createVisitor();
+			const room = await createLivechatRoom(user.token);
```
Comment (murtaza98) [inline apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts:311]: same here

Comment (KevLehman) [reply apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts:311]: Hmm, not sure i want to take the room. I just want to create it, and that's one call less 😬 on the test

Comment (murtaza98) [reply apps/meteor/tests/end-to-end/api/livechat/11-livechat.ts:311]: That's fair :+1:

**Review by murtaza98 — APPROVED** [review]

**Review by murtaza98 — DISMISSED** [review]

**Review by murtaza98 — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**